auth-backends/src/main/scala/bootstrap/rudder/plugin/AuthBackendsConf.scala

```diff
@@ -484,13 +499,68 @@ trait RudderUserServerMapping[R <: OAuth2UserRequest, U <: OAuth2User, T <: Rudd
 
 }
 
-class RudderOidcUserService(rudderUserDetailsService: RudderInMemoryUserDetailsService)
-    extends OidcUserService with RudderUserServerMapping[OidcUserRequest, OidcUser, RudderUserDetail with OidcUser] {
+class RudderOidcUserService(
+    rudderUserDetailsService: RudderInMemoryUserDetailsService,
+    registrationRepository:   RudderClientRegistrationRepository
+) extends OidcUserService with RudderUserServerMapping[OidcUserRequest, OidcUser, RudderUserDetail with OidcUser] {
   // we need to use our copy of DefaultOAuth2UserService to log/manage errors
   super.setOauth2UserService(new RudderDefaultOAuth2UserService())
 
   override def loadUser(userRequest: OidcUserRequest): OidcUser = {
-    mapRudderUser("OIDC", super.loadUser(_), rudderUserDetailsService, userRequest, new RudderOidcDetails(_, _))
+    def buildUser(oidc: OidcUser, rudder: RudderUserDetail): RudderOidcDetails = {
+      val roles = {
+        registrationRepository.registrations.get(userRequest.getClientRegistration.getRegistrationId) match {
+          case None      =>
+            AuthBackendsLogger.trace(
+              s"No configuration found for OIDC registration id: ${userRequest.getClientRegistration.getRegistrationId}"
+            )
+            rudder.roles // if no registration, use user roles
+          case Some(reg) =>
+            if (reg.roles.enabled) {
+              val custom = {
+                try {
+                  import scala.jdk.CollectionConverters._
+                  oidc
+                    .getAttribute[java.util.ArrayList[String]](reg.roles.attributeName)
+                    .asScala
+                    .map(r => RudderRoles.findRoleByName(r).runNow)
+                    .flatten
+                    .toSet
+                } catch {
+                  case ex: Exception =>
+                    AuthBackendsLogger.warn(
+                      s"Unable to get custom roles for user '${rudder.getUsername}' when looking for attribute '${reg.roles.attributeName}' :${ex.getMessage}'"
+                    )
+                    Set.empty[Role]
+                }
+              }
+
+              if (custom.nonEmpty) {
+                ApplicationLoggerPure.Authz.logEffect.info(
+                  s"Principal '${rudder.getUsername}' role list extended with OIDC provided roles: [${custom.toList.map(_.name).sorted.mkString(", ")}] (override: ${reg.roles.over})"
+                )
+              }
+
+              val roles = if (reg.roles.over) {
+                // override means: don't use user role configured in rudder-users.xml
+                custom
```

Comment:
perhaps we still need to give the `user` role which is added automatically to every users, else it might breaks some assumption. But on the other hand, adding it automatically breaks the assumption "override means that you get only the role in the list provided by the idP"